### DISCUSSION TITLE
220509 디우 인터뷰 피드백

### DISCUSSION THREAD

**Discussion by YJGwon** [woowacourse-study/2022-lv2-effective-java-interview#4] [General]
### 인터뷰어 -> 인터뷰이

> 점수를 매겨주셔도 좋고, 체크 형식으로 작성해주신 뒤 짧은 코멘트를 남겨주셔도 좋습니다. 해당 항목에 대해 자유롭게 의견을 남겨주세요.

- 말하기 태도, 어조가 적절했나요?
- 전달하고자 하는 바를 명료하게 전달했나요?
- 본인만의 주관이 잘 드러났나요?
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 모르는 부분에 대해 잘 대처했나요?

### 인터뷰이 -> 인터뷰어

> 인터뷰어로써의 역량을 키우려는 것이 아닌 저희 스터디의 원활한 진행을 위한 피드백입니다. 각 크루가 아닌 전반적인 진행 상황에 대해 의견을 남겨주세요!

- 사전에 공유한 내용이 잘 전달되었나요?
- 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- 새로운 관점을 배울 수 있었나요?

**Comment by yeon-06** [comment]
했던 질문들~
1. lock이 걸린 데이터에 다른 스레드가 접근하면 거부되는가? 아니면 순서를 대기하는가?
2. volatile를 사용하는 경우?  
  => 단순히 자원이 공유되는 것과 volatile을 붙이는 것과 어떤 것이 다른지 
3. AtomicLong에서 어떻게 thread-safe함을 지원해주는지 (ex: 내부에서 synchronized 키워드를 사용하는건지?)

자신의 경험으로 녹아내려고 한게 좋았습니당ㅎㅎ 면접 같이 질문에 대한 대답을 할때는 결론부터 말하고 사례를 구체적으로 말하는게 좋지만, 발표 같이 모르는 사람에게 새로운 개념을 설명할때는 이론 -> 사례 -> 이론 정리 식으로 설명하면 좋을 것 같아요 오늘 많은 일이 있었던 영향이 크겠지만.. 자신있게! 파이팅입니다^^~~~

**Comment by ulimy** [comment]
- 자료에 적혀있는 것을 읽을 때에는 자신감 있어보이나, 적혀있지 않은 예시 혹은 다른 이야기를 할 때 급격히 이야기가 느려지거나, 망설이는 것 같았습니다.
- 자신이 잘 모르는 기술이라도 만약 자신이 구현한다면 어떻게 할 것인지 이야기해주셔서 좋았습니다.
- 답변을 두괄식으로 해주신다면 듣기에 더 좋을 것 같아요!
- 과거에 자신이 잘못 선택했다고 생각하는 부분에 대해 자책하시는 것은 좋으나, 어느 시점이 지나면 이를 온전히 인정하고 당당해지셨으면 좋겠습니다! 인터뷰 내내 속상해하시는 것이 느껴졌어요ㅠ

**Comment by YJGwon** [comment]
- [ ] 말하기 태도, 어조가 적절했나요?
    - 레벨인터뷰때도 긴장한게 하나도 티가 안나서 놀랐었는데, 오늘은 멘탈이 깨진게 느껴졌음 ㅠ
- [X] 전달하고자 하는 바를 명료하게 전달했나요?
    - 본인이 어디까지 어떻게 이해하고 있는지 잘 전달했다고 생각합니다
- [X] 본인만의 주관이 잘 드러났나요?
   - 주관은 ㅎ... 잘 드러났습니다 디우 화이팅
- [ ] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
   - 책을 읽었다고 해서 그 항목을 다 알게되었다고 말하기 힘들다는 점을 오늘 많이 느꼈을거라고 생각해요. 공주님 화이링~!~!
- [X] 모르는 부분에 대해 잘 대처했나요?
   - 말하는 모습도 무난했지만 그보다는 정말로 배우려고 하는 태도가 보여서 기특하네유
 

미안하다는 말을 참기 힘들다면 고맙다는 말로 바꿔보세여~~
- ex)주제 선정에 실수가 있어서 죄송합니다 -> 주제 선정이 미숙했는데도 잘 진행될 수 있게 해주셔서 감사합니다

**Comment by tco0427** [comment]
- [x] 사전에 공유한 내용이 잘 전달되었나요?
- [ ] 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
    -> 본인이 준비한 내용이 어떤 의견을 끌어내기보다는 `이펙티브 자바`를 단순 정리한 것 같아서 인터뷰어분들이 어떤 의견을 끌어내기는 어려웠던 것 같아요ㅠ.ㅠ
- [x] 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
    -> "AtomicXXX 가 어떻게 동작하는가?", `Synchronized` 와 `volatile` 의 차이등 보다 깊은 공부를 할 수 있는 질문을 주셔서 감사했습니다.
- [x] 새로운 관점을 배울 수 있었나요?
    -> 완전..

**Comment by kyukong** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [ ] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

레벨로그 때 보니까 발표를 너무 잘했었는데 잘못된 아이템 선정때문인지 후회가 많아보였습니다.ㅋㅋ 이펙티브 자바 내용만이라도 전달해주셨으면 좋았을 것 같습니다. 그래도 덕분에 스레드에 대해 조금이라도 공부해볼 수 있었어요.

**Comment by Ohzzi** [comment]
- 말하기 태도, 어조가 적절했나요?
  - 시작 전에 주제를 잘못 잡았다고 인지하고 왔어도 인터뷰 때는 당당하게 하기!
- 전달하고자 하는 바를 명료하게 전달했나요?
  - 전달하고자 하는 바 자체는 명료하게 전달되었다고 생각합니다!
- 본인만의 주관이 잘 드러났나요?
  - 잘 드러났습니다 :) 
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
  - 책의 내용을 무작정 수용하지 않으신 부분 좋습니다 :)
- 모르는 부분에 대해 잘 대처했나요?
  - volatile, synchronized같이 결국 책에는 나온 내용이지만 실제 프로젝트에 적용 안하신 부분인데 이 부분에 대한 대처가 살짝 부족했다고 느낍니다!
